File: TOMIMSV4/Entity/Transacciones/Recepcion/Recepcion_OC/clsBeTrans_re_oc.vb
Public Class clsBeTrans_re_oc
    Implements ICloneable
    Implements IDisposable

    Public Property IdRecepcionOc() As Integer = 0
    Public Property IdRecepcionEnc() As Integer = 0
    Public Property IdOrdenCompraEnc() As Integer = 0
    Public Property Recepcion_ciega() As Boolean = False
    Public Property Recepcion_manual() As Boolean = False
    Public Property No_docto() As String = ""
    Public Property Hora_ini_hh() As Date = Date.Now
    Public Property Hora_fin_hh() As Date = Date.Now
    Public Property User_agr() As String = ""
    Public Property Fec_agr() As Date = Date.Now
    Public Property Firma_operador() As Byte() = Nothing

    Sub New()
    End Sub

    Sub New(ByRef IdRecepcionOc As Integer, ByVal IdRecepcionEnc As Integer, ByVal IdOrdenCompraEnc As Integer, ByVal recepcion_ciega As Boolean, ByVal recepcion_manual As Boolean, ByVal no_docto As String, ByVal hora_ini_hh As Date, ByVal hora_fin_hh As Date, ByVal user_agr As String, ByVal fec_agr As Date, ByVal firma_operador As Byte())
        Me.IdRecepcionOc = IdRecepcionOc
        Me.IdRecepcionEnc = IdRecepcionEnc
        Me.IdOrdenCompraEnc = IdOrdenCompraEnc
        Me.Recepcion_ciega = recepcion_ciega
        Me.Recepcion_manual = recepcion_manual
        Me.No_docto = no_docto
        Me.Hora_ini_hh = hora_ini_hh
        Me.Hora_fin_hh = hora_fin_hh
        Me.User_agr = user_agr
        Me.Fec_agr = fec_agr
        Me.Firma_operador = firma_operador
    End Sub

    Public Function Clone() As Object Implements System.ICloneable.Clone
        Return MyBase.MemberwiseClone()
    End Function

#Region "IDisposable Support"
    Private disposedValue As Boolean ' To detect redundant calls

    ' IDisposable
    Protected Overridable Sub Dispose(disposing As Boolean)
        If Not disposedValue Then
            If disposing Then
                ' TODO: dispose managed state (managed objects).
            End If

            ' TODO: free unmanaged resources (unmanaged objects) and override Finalize() below.
            ' TODO: set large fields to null.
        End If
        disposedValue = True
    End Sub

    ' TODO: override Finalize() only if Dispose(disposing As Boolean) above has code to free unmanaged resources.
    'Protected Overrides Sub Finalize()
    '    ' Do not change this code.  Put cleanup code in Dispose(disposing As Boolean) above.
    '    Dispose(False)
    '    MyBase.Finalize()
    'End Sub

    ' This code added by Visual Basic to correctly implement the disposable pattern.
    Public Sub Dispose() Implements IDisposable.Dispose
        ' Do not change this code.  Put cleanup code in Dispose(disposing As Boolean) above.
        Dispose(True)
        ' TODO: uncomment the following line if Finalize() is overridden above.
        ' GC.SuppressFinalize(Me)
    End Sub
#End Region

End Class

File: WMS.EntityCore/Trans_re/clsBeTrans_re_oc.cs
using System.ComponentModel;
using System.ComponentModel.DataAnnotations.Schema;
using WMS.EntityCore.Trans_oc;

namespace WMS.EntityCore.Trans_re
{
    public class clsBeTrans_re_oc : ICloneable
    {
        [Column("IdRecepcionOc")]
        [DisplayName("IdRecepcionOc")]
        public int IdRecepcionOc { get; set; } = 0;
        [Column("IdRecepcionEnc")]
        [DisplayName("IdRecepcionEnc")]
        public int IdRecepcionEnc { get; set; } = 0;
        [Column("IdOrdenCompraEnc")]
        [DisplayName("IdOrdenCompraEnc")]
        public int IdOrdenCompraEnc { get; set; } = 0;
        [Column("recepcion_ciega")]
        [DisplayName("recepcion_ciega")]
        public bool Recepcion_ciega { get; set; } = false;
        [Column("recepcion_manual")]
        [DisplayName("recepcion_manual")]
        public bool Recepcion_manual { get; set; } = false;
        [Column("no_docto")]
        [DisplayName("no_docto")]
        public string No_docto { get; set; } = "";
        [Column("hora_ini_hh")]
        [DisplayName("hora_ini_hh")]
        public DateTime Hora_ini_hh { get; set; } = DateTime.Now;
        [Column("hora_fin_hh")]
        [DisplayName("hora_fin_hh")]
        public DateTime Hora_fin_hh { get; set; } = DateTime.Now;
        [Column("user_agr")]
        [DisplayName("user_agr")]
        public string User_agr { get; set; } = "";
        [Column("fec_agr")]
        [DisplayName("fec_agr")]
        public DateTime Fec_agr { get; set; } = DateTime.Now;
        [Column("firma_operador")]
        [DisplayName("firma_operador")]
        public byte[] Firma_operador { get; set; } = Array.Empty<byte>();
        public bool IsNew { get; set; } = false;
        public clsBeTrans_oc_enc OC { get; set; } = new clsBeTrans_oc_enc();
        public clsBeTrans_re_oc() { }
        public object Clone()
        {
            return MemberwiseClone();
        }
    }
}
